src/libopensc/card-iasecc.c

```diff
@@ -594,6 +609,36 @@ iasecc_init_amos_or_sagem(struct sc_card *card)
 }
 
 
+static int
+iasecc_init_cpx(struct sc_card *card)
+{
+	struct sc_context *ctx = card->ctx;
+	unsigned int flags; /* TBC it is not IASECC_CARD_DEFAULT_FLAGS */
+
+	LOG_FUNC_CALLED(ctx);
+
+	card->caps = SC_CARD_CAP_RNG; /* TBC it is not IASECC_CARD_DEFAULT_CAPS */
+
+	flags = SC_ALGORITHM_RSA_PAD_PKCS1;
+	flags |= SC_ALGORITHM_RSA_RAW;
+
+	/* No signature with contactless mode */
```

Comment:
I think what you do here does not reflect this comment. What you do allows raw RSA and RSA with PKCS1 padding. If you do not want to allow any signatures in contactless mode, you should not set flags at all and not call `_sc_card_add_rsa_alg`.

Reply:
Right, currently, I am focusing on the Contact mode. The contactless mode is supposed to be close to the contactless mode, but with a different AID and no PIN code. For the time being, I'll focus on getting a full support of the contact mode.

So let's focus on this scope:
   https://github.com/OpenSC/OpenSC/pull/2217/commits/3867bd357a9122f78aa92bfa4a49c584e4e47b7c